compiler/luci/pass/src/FuseHorizontalFCLayers.cpp

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseHorizontalFCLayers.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Service/Nodes/CircleConst.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+
+namespace luci
+{
+
+bool check_type_and_shape_equality(const CircleNode *left, const CircleNode *right)
+{
```

Comment:
optional) usually we put local methods inside anonymous namespace
```suggestion

namespace
{

bool check_type_and_shape_equality(const CircleNode *left, const CircleNode *right)
{
```

may have to add closing of anonymous namespace closing below.

Reply:
Done